### PR TITLE
Remove deprecated shells.

### PR DESCRIPTION
**Pull request by ADmad**
Phew! That was a lot of work.

### CODE REVIEW

File: src/Console/Command/HelpCommand.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ protected function asText(ConsoleIo $io, iterable $commands): void
         $grouped = [];
         $plugins = Plugin::loaded();
         foreach ($invert as $class => $names) {
-            preg_match('/^(.+)\\\\(Command|Shell)\\\\/', $class, $matches);
+            preg_match('/^(.+)\\\\(Command)\\\\/', $class, $matches);
```
Comment (garas) [inline src/Console/Command/HelpCommand.php:98]:
```suggestion
            preg_match('/^(.+)\\\\Command\\\\/', $class, $matches);
```

File: tests/TestCase/Console/CommandCollectionTest.php [hunk 2]
```diff
@@ -123,18 +122,6 @@ public function testAddInstance()
...
-    /**
-     * Instances that are not shells should fail.
-     */
-    public function testAddInvalidInstance()
-    {
-        $this->expectException(\InvalidArgumentException::class);
-        $this->expectExceptionMessage('Cannot use \'stdClass\' for command \'routes\'. It is not a subclass of Cake\Console\Shell');
-        $collection = new CommandCollection();
-        $shell = new stdClass();
-        $collection->add('routes', $shell);
```
Comment (othercorey) [inline tests/TestCase/Console/CommandCollectionTest.php:135]: Is this possible with commands?

File: tests/TestCase/Console/CommandRunnerTest.php [hunk 3]
```diff
@@ -311,27 +311,12 @@ public function testRunValidCommandInflection()
...
-        $result = $runner->run(['cake', 'failure', 'with_abort'], $this->getMockIo($output));
-        $this->assertSame(Shell::CODE_ERROR, $result);
-    }
-
-    /**
-     * Test returning a non-zero value
-     *
-     * @return void
-     */
-    public function testRunValidCommandReturnInteger()
```
Comment (othercorey) [inline tests/TestCase/Console/CommandRunnerTest.php:327]: Is this possible with commands?

File: tests/TestCase/Console/CommandCollectionTest.php (hunk 2 above)
Comment (ADmad) [reply tests/TestCase/Console/CommandCollectionTest.php:135]: The 2nd argument now has union type `string|CommandInterface` so this test is no longer necessary.

File: tests/TestCase/Console/CommandRunnerTest.php (hunk 3 above)
Comment (ADmad) [reply tests/TestCase/Console/CommandRunnerTest.php:327]: There's already another test which with integer return value.

**Comment by othercorey** [comment]
@markstory Looks ok, but I'm not familiar with everything.

**Review by markstory — APPROVED** [review]
Nice :clap:

**Pull request merged**